pkg/reconciler/tenancy/clusterworkspace/deletion/workspace_resource_deletor.go

```diff
@@ -0,0 +1,536 @@
+/*
```

Comment:
how much of this file is from namespace deletion?

Reply:
I think 70%. I remove the opCache part, and change how conditions are updated. Add a comment, and also add kubernetes copyright in this file.